flowey/flowey_lib_common/src/gh_workflow_id.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Gets the Github workflow id for a given commit hash
+
+use flowey::node::prelude::*;
+
+flowey_request! {
+    pub struct Request {
+        pub github_commit_hash: ReadVar<String>,
+        pub repo_path: ReadVar<PathBuf>,
+        pub gh_workflow_id: WriteVar<String>,
+        pub pipeline_name: String,
+        pub gh_token: ReadVar<String>,
+    }
+}
+
+new_simple_flow_node!(struct Node);
+
+impl SimpleFlowNode for Node {
+    type Request = Request;
+
+    fn imports(ctx: &mut ImportCtx<'_>) {
+        ctx.import::<crate::use_gh_cli::Node>();
+    }
+
+    fn process_request(request: Self::Request, ctx: &mut NodeCtx<'_>) -> anyhow::Result<()> {
+        let Request {
+            repo_path,
+            github_commit_hash,
+            gh_workflow_id,
+            pipeline_name,
+            gh_token,
+        } = request;
+
+        let pipeline_name = pipeline_name.clone();
+
+        ctx.req(crate::use_gh_cli::Request::WithAuth(
+            crate::use_gh_cli::GhCliAuth::AuthToken(gh_token.clone()),
+        ));
+        let gh_cli = ctx.reqv(crate::use_gh_cli::Request::Get);
+
+        ctx.emit_rust_step("get action id", |ctx| {
+            let gh_workflow_id = gh_workflow_id.claim(ctx);
+            let github_commit_hash = github_commit_hash.claim(ctx);
+            let gh_token = gh_token.claim(ctx);
+            let repo_path = repo_path.claim(ctx);
+            let pipeline_name = pipeline_name.clone();
+            let gh_cli = gh_cli.claim(ctx);
+
+            move |rt| {
+                let github_commit_hash = rt.read(github_commit_hash);
+                let sh = xshell::Shell::new()?;
+                let gh_token = rt.read(gh_token);
+                let repo_path = rt.read(repo_path);
+                let gh_cli = rt.read(gh_cli);
+
+                sh.change_dir(repo_path);
+
+                // Fetches the CI build workflow id for a given commit hash
+                let get_action_id = |commit: String| {
+                    sh.cmd(gh_cli.clone())
```

Comment:
kinda feels like you swung a bit too far ino the opposite direction here - this massive arg chain seems worse than just invoking `xtask::cmd!`, no?